lib/livebook/session.ex

```diff
@@ -410,6 +418,11 @@ defmodule Livebook.Session do
     {:reply, summary_from_state(state), state}
   end
 
+  def handle_call(:get_notebook_source, _from, state) do
+    content = LiveMarkdown.Export.notebook_to_markdown(state.data.notebook)
+    {:reply, content, state}
```

Comment:
I am wondering if we should return the data and do the conversion on the client to avoid blocking the server?

Reply:
Good call, copying is cheaper. Also having the data we can export it into any other format without involving the session!